include/zephyr/dt-bindings/clock/stm32f10x_clock.h

```diff
@@ -72,4 +73,8 @@
 /** BDCR devices */
 #define RTC_SEL(val)		STM32_CLOCK(val, 3, 8, BDCR_REG)
 
+#define MCO1_SEL(val)           STM32_MCO_CFGR(val, 0x7, 24, CFGR1_REG)
+/* Prescaler not supported. */
```

Comment:
I think the intent is to make explicit that "`MCO1_PRE` is not present because the feature is not available, not because it was forgotten".

Reply:
This series does not have a MCOPRE in CFGR register, the mask is set to zero.

Reply:
Should we improve on the comment? Happy to take suggestions here.

Reply:
Proposal:
```suggestion
/* STM32F1 Connectivity line has no MCO prescaler. */
```

Although, I see now that the `prescaler` property is marked as `required: true`, so we have to handle this special case.

I see two solutions:
* remove the `required` attribute from the property, and use `DT_PROP_OR(node, prescaler, /* DRIVER_SENTINEL_DEFINE */)` when obtaining prescaler
  * when applying the configuration, only write the prescaler if `config->prescaler != /* DRIVER_SENTINEL_DEFINE */`
  * define could be something like `#define MCO_DONT_CONFIGURE_PRESCALER 0xFFFFFFFF` in `clock_stm32_mco.c`
* keep the `required` attribute on the property, but define a (universal) fallback value (e.g., in `stm32_clocks_common.h`, `#define STM32_MCO_CFGR_NONE 0xFFFFFFFF`)
  * in `stm32f10x_clock.h`, we'd have a single `#define MCO1_NO_PRESCALER STM32_MCO_CFGR_NONE` define in place of `MCO_PRE`
  * driver would still only apply configuration if `config->prescaler != STM32_MCO_CFGR_NONE`

I would lean towards the second solution, which could give something easy to understand like:
```suggestion
/* STM32F1 Connectivity line has no MCO prescaler.
 * Use the following as the `prescaler` property for MCO1:
 */
#define MCO1_PRESCALER_NONE STM32_MCO_CFGR_NONE
```

It's also safer as it guarantees the user has explicitly chosen the prescaler - no possible bad surprise.

CC'ing @erwango for input

Reply:
To me, the canonical way (maybe not the most simple) to deal with this is:
- dts/bingings: create a new `st,stm32f1-mco` compatible which includes `st,stm32-mco` but blocks `prescaler` property.
- dts/arm/st... use the correct compatible
- driver/clock_control
 - Kconfig: depends on DT_HAS_ST_STM32_MCO_ENABLED || DT_HAS_ST_STM32F1_MCO_ENABLED
 - ...._mco.c: 
   - Define `DT_DRV_COMPAT` conditionally
```
 #if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_mco)
#define DT_DRV_COMPAT st_stm32_mco
#else
#define DT_DRV_COMPAT st_stm32f1_mco
#endif
```    
   - Encapsulate the prescaler setting under `#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_mco)`
   - Define prescaler with 
```
	IF_ENABLED(DT_HAS_COMPAT_STATUS_OKAY(st_stm32_mco),		\
		(.prescaler = DT_INST_PROP(id, prescaler),))		\
```

Reply:
Personally I think leaving the prescaler as not required would not be a bad idea.

If it is not used, then we can just leave it at its register default value, this can be added to the property description.
I think across all devices the reset value of the CFGR register results in a prescaler division of 1 i.e. no division.
If I understand correctly this is Mathiu's first option.

We can update the sample to also demonstrate this, for example on the discovery board which has 2 MCOs, we can have one with prescaler, and one without.

In the driver we throw an error for the devices that don't have prepscaler support if this is added.
This can be done by checking for Kconfig F1xx series, or we can have devicetree property like: `no-prescaler-support` added on this series.

@erwango  I guess you're the one to make a decision?
Even with your approach we may also consider leaving prescaler as optional and just leave it at reset value.
Two things to be decided:
1. How or if to block prescaler configuration on f1xx series.
2. Should prescaler be required.

Reply:
> Personally I think leaving the prescaler as not required would not be a bad idea.

This is what Erwan is proposing, but using a specific compatible for the lines/series with such quirk (i.e., STM32F1).

This way, F1 DTS will not require `prescaler` (and, in fact, error out if provided), but all other series will. This ensures that on all series, the MCO is fully configured by DTS (and thus user) before being enabled.

> If it is not used, then we can just leave it at its register default value, this can be added to the property description.
I think across all devices the reset value of the CFGR register results in a prescaler division of 1 i.e. no division.
If I understand correctly this is Mathiu's first option.

Actually, there are exceptions to this - for example, on STM32H7 (RM0399): `MCO2 prescaler [...] 0000: prescaler disabled (default after reset)` - i.e., default prescaler configuration is *disable MCO*, not *divide by 1*.

> How or if to block prescaler configuration on f1xx series.

If you mean, how to declare it in a binding, see e.g.: https://github.com/zephyrproject-rtos/zephyr/blob/494f88070f6bb909389630264e4f89c71072ae53/dts/bindings/clock/st%2Cstm32f100-pll-clock.yaml#L25-L28

In our case, it would be `st,stm32f1-mco.yaml` containing:
```yaml
 include: 
   - name: st,stm32-mco.yaml 
     property-blocklist: 
       - prescaler
```

cc @erwango

Reply:
> This is what Erwan is proposing

Exactly:
- we let dt thow errors on misconfigurations (eg prescaler on non compatible F1 devices)
- in the driver we only deal with correct configurations

Reply:
Alright, going for that approach then.